stream-platform/src/main/java/io/camunda/zeebe/stream/api/InterPartitionCommandSender.java

```diff
@@ -26,10 +26,8 @@ void sendCommand(
    * Uses the given record key when writing the command. Otherwise, behaves like {@link
    * InterPartitionCommandSender#sendCommand}
    *
-   * @deprecated This is only available for compatability with deployment distribution.
    * @param recordKey Record key to use when writing the command. Ignored if null.
    */
-  @Deprecated(forRemoval = true)
```

Comment:
❌ We (i.e. I) promised to document clearly when it would be okay to use this method. (see archived slack channel #top-zeebe-dedeprecate-send-command-with-key). 

Something along the lines of:

> Uses the given record key when writing the command. This may be used in cases where the key has been provided to the users, and the entity must be available on the receiving partition with that specific key. This also helps inform the receiving partition that this command was sent from another partition (and from which partition as the key encodes the partition id).